instrumentation/lettuce-4.3/src/main/java/com/lambdaworks/redis/StatefulRedisConnectionImpl.java

```diff
@@ -0,0 +1,22 @@
+package com.lambdaworks.redis;
+
+import java.util.concurrent.TimeUnit;
+
+import com.lambdaworks.redis.RedisURI;
+import com.lambdaworks.redis.api.StatefulConnection;
+import com.lambdaworks.redis.codec.RedisCodec;
+import com.newrelic.api.agent.weaver.MatchType;
+import com.newrelic.api.agent.weaver.NewField;
+import com.newrelic.api.agent.weaver.Weave;
+
+@Weave(type = MatchType.BaseClass)
+public abstract class StatefulRedisConnectionImpl<K, V> implements StatefulConnection<K, V> {
```

Comment:
Best practice is to name any instrumented library classes with `_Instrumentation` appended to the original class name (e.g. `StatefulRedisConnectionImpl_Instrumentation<K, V>`) and then specify the original class name using `@Weave(type = MatchType.BaseClass, originalName="com.lambdaworks.redis.StatefulRedisConnectionImpl<K, V>")`.

This was done correctly for `RedisClient_Instrumentation` and we should be consistent about that.